File: 프로그래머스 문제/level2/짝지어 제거하기.py
def solution(s):
    stack = []

    for w in s:
        if not stack:
            stack.append(w)
        else:
            if stack[-1] == w:
                stack.pop()
            else:
                stack.append(w)
    if not stack:
        return 1
    else:
        return 0

print(solution('baabaa'))
print(solution('cdcd'))

File: js연습/level2/짝지어 제거하기.js
function solution(s)
{
  if (s.length % 2 != 0) {
    return 0;
  }
  var answer = -1;
  let stack = [];
  for (let i = 0; i < s.length; i++) {
     if (stack.length == 0) {
      stack.push(s[i])
     } else {
      stack.at(-1) === s[i] ? stack.pop() : stack.push(s[i])
     }
  }

  stack.length == 0 ? answer = 1 : answer = 0

  return answer;
}